main/src/cgeo/geocaching/EditWaypointActivity.java

```diff
@@ -333,29 +333,29 @@ public void onNothingSelected(final AdapterView<?> parent) {
             }
         });
 
-        binding.type.setSelection(getDefaultWaypointType());
+        binding.type.setSelection(getDefaultWaypointTypePosition(wpAdapter));
         binding.type.setVisibility(View.VISIBLE);
     }
 
-    private int getDefaultWaypointType() {
+    private int getDefaultWaypointTypePosition(final ArrayAdapter<WaypointType> wpAdapter) {
         // potentially restore saved instance state
         if (waypointTypeSelectorPosition >= 0) {
             return waypointTypeSelectorPosition;
         }
 
         // when editing, use the existing type
         if (waypoint != null) {
-            return POSSIBLE_WAYPOINT_TYPES.indexOf(waypoint.getWaypointType());
```

Comment:
Where else is this "POSSIBLE_WAYPOINT_TYPES" constant used? Can it be deleted now? If not: might this change cause conflicts with places where it is still in use?

Reply:
The array is used to initialize the spinner, depending on wheter it is a generated waypoint POSSIBLE_WAYPOINT_TYPES or POSSIBLE_WAYPOINT_TYPES_WITH_GENERATED is used.

Reply:
ok